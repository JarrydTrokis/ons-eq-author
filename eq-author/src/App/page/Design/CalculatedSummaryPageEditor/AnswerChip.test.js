import React from "react";
import { shallow } from "enzyme";

import AnswerChip from "./AnswerChip";

describe("AnswerChip", () => {
  const onRemove = jest.fn();
  it("should render", () => {
    const wrapper = shallow(<AnswerChip onRemove={onRemove}>Hello</AnswerChip>);
    expect(wrapper).toMatchSnapshot();
  });
  it("should call onRemove on x click", () => {
    const wrapper = shallow(<AnswerChip onRemove={onRemove}>Hello</AnswerChip>);
    wrapper.find("[data-test='remove-answer-button']").simulate("click");
    expect(onRemove).toHaveBeenCalled();
  });
});
