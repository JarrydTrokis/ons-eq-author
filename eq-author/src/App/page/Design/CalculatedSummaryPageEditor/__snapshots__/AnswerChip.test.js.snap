// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerChip should render 1`] = `
<AnswerChip__Chip>
  <AnswerChip__Title>
    Hello
  </AnswerChip__Title>
  <AnswerChip__CloseButton
    data-test="remove-answer-button"
    onClick={[MockFunction]}
  />
</AnswerChip__Chip>
`;
